query/src/storages/random/random_table.rs

```diff
@@ -0,0 +1,188 @@
+    ) -> Result<(Statistics, Partitions)> {
+        Ok((Statistics::default(), vec![]))
+    }
+
+    fn benefit_column_prune(&self) -> bool {
+        true
+    }
+
+    fn read2(
+        &self,
+        ctx: Arc<QueryContext>,
+        plan: &ReadDataSourcePlan,
+        pipeline: &mut NewPipeline,
+    ) -> Result<()> {
+        let output = OutputPort::create();
+        let schema = self.table_info.schema();
+        pipeline.add_pipe(NewPipe::SimplePipe {
+            inputs_port: vec![],
+            outputs_port: vec![output.clone()],
+            processors: vec![RandomSource::create(
+                ctx,
+                output,
+                schema,
+                plan.push_downs.clone(),
+            )?],
+        });
+
+        Ok(())
+    }
+}
+
+struct RandomSource {
+    schema: DataSchemaRef,
+    /// The number of rows needed to generate.
+    limit: usize,
+    /// record rows count.
+    rows: usize,
+    /// The max number of rows can one `generate` generate.
+    block_size: usize,
+}
+
+impl RandomSource {
+    pub fn create(
+        ctx: Arc<QueryContext>,
+        output: Arc<OutputPort>,
+        schema: DataSchemaRef,
+        extras: Option<Extras>,
+    ) -> Result<ProcessorPtr> {
+        let block_size = ctx.get_settings().get_max_block_size()? as usize;
+        let mut output_schema = schema;
+        // If extras.push_downs is None or extras.push_down.limit is None,
+        // set limit to `max_block_size`.
+        let limit = match extras {
+            Some(push_downs) => {
+                if let Some(projection) = push_downs.projection {
+                    output_schema = Arc::new(output_schema.project(projection));
+                }
+                match push_downs.limit {
+                    Some(limit) => limit,
+                    None => block_size,
+                }
+            }
+            None => block_size,
+        };
+
+        SyncSourcer::create(ctx, output, RandomSource {
+            schema: output_schema,
+            limit,
+            rows: 0,
+            block_size,
+        })
+    }
+}
+
+impl SyncSource for RandomSource {
+    const NAME: &'static str = "RandomTable";
+
+    fn generate(&mut self) -> Result<Option<DataBlock>> {
+        if self.rows == self.limit {
+            return Ok(None);
+        }
+
+        let mut packs = self
+            .schema
+            .fields()
+            .iter()
+            .map(|f| f.data_type().create_deserializer(self.block_size))
+            .collect::<Vec<_>>();
+
+        let format = FormatSettings::default();
+
+        let mut rows = 0;
+
+        loop {
+            for (col, pack) in packs.iter_mut().enumerate() {
+                let col_data_type = self.schema.field(col).data_type();
+                pack.append_data_value(col_data_type.random_value(), &format)?;
+            }
+
+            rows += 1;
+            self.rows += 1;
+
+            // Only generate at most `block_size` rows in one turn.
```

Comment:
I don't know how to generate rows with exactly `block_size` of physical memory size, so I simply generate `block_size` rows one time. Is this right?